model/jpa/src/main/java/org/keycloak/models/jpa/session/JpaUserSessionPersisterProvider.java

```diff
@@ -239,6 +241,52 @@ public void removeExpired(RealmModel realm) {
 
     }
 
+    @Override
+    public UserSessionModel loadUserSession(String userSessionId, boolean offline) {
+
+        String offlineStr = offlineToString(offline);
+
+        TypedQuery<PersistentUserSessionEntity> userSessionQuery = em.createNamedQuery("findUserSession", PersistentUserSessionEntity.class);
+        userSessionQuery.setParameter("offline", offlineStr);
+        userSessionQuery.setParameter("userSessionId", userSessionId);
+        userSessionQuery.setMaxResults(1);
+
+        List<PersistentUserSessionAdapter> result = userSessionQuery.getResultStream()
+                .map(this::toAdapter)
+                .collect(Collectors.toList());
+
+        if (result.isEmpty()) {
+            return null;
+        }
+
+        PersistentUserSessionAdapter userSession = result.get(0);
+
+        TypedQuery<PersistentClientSessionEntity> clientSessionQuery = em.createNamedQuery("findClientSessionsByUserSessions", PersistentClientSessionEntity.class);
```

Comment:
Isn't possible to reuse the same code from `loadUserSessions` method?

Reply:
Thanks for your feedback @pedroigor 

> org.keycloak.models.jpa.session.JpaUserSessionPersisterProvider#loadUserSessions
That's what I tried at first, but it turned out that, the session referenced by the `lastUserSessionId` parmeter is not included in the result... that's why I decided to implement the new method.

Reply:
... of course we could change the `sess.userSessionId > :lastSessionId` condition in the findUserSessions JPA query to `sess.userSessionId >= :lastSessionId`, but I'd also need to adjust the other conditions or pass dummy values (0?) for `lastCreatedOn`.

hmmm, I still think that the two methods are fine for now.